server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -110,6 +110,31 @@ private static void forEach(Object arrayOrCollection, Consumer<Object> consumer)
         }
     }
 
+    private static Iterator<?> iterator(Object arrayOrCollection) {
+        if (arrayOrCollection.getClass().isArray()) {
+            return new Iterator<>() {
+                int arrayLength = Array.getLength(arrayOrCollection);
+                int current = 0;
+
+                @Override
+                public boolean hasNext() {
+                    return current < arrayLength;
+                }
+
+                @Override
+                public Object next() {
+                    int c = current;
+                    current++;
+                    return Array.get(arrayOrCollection, c);
+                }
+            };
+        } else if (arrayOrCollection instanceof Collection<?> collection) {
+            return collection.iterator();
+        } else {
+            throw new AssertionError("argument is neither an array nor a collection");
+        }
```

Comment:
Not nice to do boxing for the comparison if the input format is `double[]` on both sides.
I think this would be worth special casing.